storage/pebble/bootstrap_test.go

```diff
@@ -0,0 +1,259 @@
+	})
+}
+
+func TestRegisterBootstrap_IndexCheckpointFile_MultipleBatch(t *testing.T) {
+	t.Parallel()
+	log := zerolog.New(io.Discard)
+	rootHeight := uint64(10000)
+	unittest.RunWithTempDir(t, func(dir string) {
+		tries, registerIDs := largeTrieWithValidRegisterIDs(t)
+		fileName := "large-checkpoint"
+		require.NoErrorf(t, wal.StoreCheckpointV6Concurrently(tries, dir, fileName, log), "fail to store checkpoint")
+		checkpointFile := path.Join(dir, fileName)
+		cache := pebble.NewCache(1 << 20)
+		opts := DefaultPebbleOptions(cache, registers.NewMVCCComparer())
+		defer cache.Unref()
+		pb, dbDir := unittest.TempPebbleDBWithOpts(t, opts)
+		bootstrap, err := NewRegisterBootstrap(pb, checkpointFile, rootHeight, log)
+		require.NoError(t, err)
+		err = bootstrap.IndexCheckpointFile(context.Background())
+		require.NoError(t, err)
+
+		// create registers instance and check values
+		reg, err := NewRegisters(pb)
+		require.NoError(t, err)
+
+		require.Equal(t, reg.LatestHeight(), rootHeight)
+		require.Equal(t, reg.FirstHeight(), rootHeight)
+
+		for _, register := range registerIDs {
+			val, err := reg.Get(*register, rootHeight)
+			require.NoError(t, err)
+			require.Equal(t, val, []byte{uint8(0)})
+		}
+
+		require.NoError(t, pb.Close())
+		require.NoError(t, os.RemoveAll(dbDir))
+	})
+
+}
+
+func simpleTrieWithValidRegisterIDs(t *testing.T) ([]*trie.MTrie, []*flow.RegisterID) {
+	return trieWithValidRegisterIDs(t, 2)
+}
+
+func largeTrieWithValidRegisterIDs(t *testing.T) ([]*trie.MTrie, []*flow.RegisterID) {
+	// large enough trie so every worker should have something to index
+	largeTrieSize := 2 * pebbleBootstrapRegisterBatchLen * pebbleBootstrapWorkerCount
+	return trieWithValidRegisterIDs(t, uint16(largeTrieSize))
+}
+
+func trieWithValidRegisterIDs(t *testing.T, n uint16) ([]*trie.MTrie, []*flow.RegisterID) {
+	emptyTrie := trie.NewEmptyMTrie()
+	resultRegisterIDs := make([]*flow.RegisterID, 0, n)
+	paths := randomRegisterPaths(n)
+	payloads := randomRegisterPayloads(n)
+	for _, payload := range payloads {
+		key, err := payload.Key()
+		require.NoError(t, err)
+		regID, err := keyToRegisterID(key)
+		require.NoError(t, err)
+		resultRegisterIDs = append(resultRegisterIDs, &regID)
+	}
+	populatedTrie, depth, err := trie.NewTrieWithUpdatedRegisters(emptyTrie, paths, payloads, true)
+	// make sure it has at least 1 leaf node
+	require.GreaterOrEqual(t, depth, uint16(1))
+	require.NoError(t, err)
+	resultTries := []*trie.MTrie{emptyTrie, populatedTrie}
+	return resultTries, resultRegisterIDs
+}
+
+func randomRegisterPayloads(n uint16) []ledger.Payload {
+	p := make([]ledger.Payload, 0, n)
+	for i := uint16(0); i < n; i++ {
+		o := make([]byte, 0, 8)
+		o = binary.BigEndian.AppendUint16(o, n)
+		k := ledger.Key{KeyParts: []ledger.KeyPart{
+			{Type: state.KeyPartOwner, Value: o},
+			{Type: state.KeyPartKey, Value: o},
+		}}
+		// values are always 0 for ease of testing/checking
+		v := ledger.Value{uint8(0)}
```

Comment:
Can you update this to use a more specific value. It would be easy for 0 to accidentally work due to a bug.